src/etl/sars-cov-2/steps/fetch-positive-case-data-step.ts

```diff
@@ -0,0 +1,303 @@
+  "South Africa": "ZA",
+  "South Korea": "KR",
+  "South Sudan": "SS",
+  "Spain": "ES",
+  "Sri Lanka": "LK",
+  "Sudan": "SD",
+  "Suriname": "SR",
+  "Sweden": "SE",
+  "Switzerland": "CH",
+  "Syria": "SY",
+  "Taiwan": "TW",
+  "Tajikistan": "TJ",
+  "Tanzania": "TZ",
+  "Thailand": "TH",
+  "Timor": "TL",
+  "Togo": "TG",
+  "Tonga": "TO",
+  "Trinidad and Tobago": "TT",
+  "Tunisia": "TN",
+  "Turkey": "TR",
+  "Turks and Caicos Islands": "TC",
+  "Tuvalu": "TV",
+  "Uganda": "UG",
+  "Ukraine": "UA",
+  "United Arab Emirates": "AE",
+  "United Kingdom": "GB",
+  "United States": "US",
+  "Uruguay": "UY",
+  "Uzbekistan": "UZ",
+  "Vanuatu": "VU",
+  "Vatican": "VA",
+  "Venezuela": "VE",
+  "Vietnam": "VN",
+  "Wallis and Futuna": "WF",
+  "Yemen": "YE",
+  "Zambia": "ZM",
+  "Zimbabwe": "ZW"
+}
+
+export const fetchPositiveCaseDataStep = async(
+  input: FetchPositiveCaseDataStepInput
+): Promise<FetchPositiveCaseDataStepOutput> => {
+  console.log(`Running step: fetchPositiveCaseDataStep. Remaining estimates: ${input.allEstimates.length}`);
+
+  const { body } = await request(
+    "https://raw.githubusercontent.com/owid/covid-19-data/master/public/data/jhu/total_cases_per_million.csv",
+    { method: "GET" }
+  );
+
+  const rawCsvData = await body.text();
+  
+  const csvColumns = rawCsvData.split("\n").at(0)?.split(',') ?? []
+  const indexOfDateColumn = csvColumns.findIndex(columnValue => columnValue === 'date');
+  const countryCodesWithIndices = typedObjectEntries(countryNamesInCSVToTwoLetterCountryCode)
+    .map(([countryNameInCsvFile, twoLetterCountryCode]) => ({
+      twoLetterCountryCode: twoLetterCountryCode,
+      indexOfColumn: csvColumns.findIndex(columnValue => columnValue === countryNameInCsvFile)
+    }))
+
+  const unformattedPositiveCaseData = rawCsvData
+    .split("\n")
+    .flatMap((element, index) => {
+      if (index === 0 || !element) {
+        return undefined;
+      }
+
+      const split_csv_line = element.split(",");
+      const date = split_csv_line[indexOfDateColumn];
+      const year = date.split("-")[0]
+      const month = date.split("-")[1]
+      const day = date.split("-")[2]
+
+      return countryCodesWithIndices.map(({ twoLetterCountryCode, indexOfColumn }) => (twoLetterCountryCode ? {
+        twoLetterCountryCode,
+        year,
+        month,
+        day,
+        countryPositiveCasesPerMillionPeople: split_csv_line[indexOfColumn] !== '' ? parseFloat(split_csv_line[indexOfColumn]) : undefined,
+      } : undefined)).filter(<T>(element: T | undefined): element is T => !!element);
+    })
+    .filter(<T>(element: T | undefined): element is T => !!element);
+
+  const formattedPositiveCaseData = groupByArray(
+    unformattedPositiveCaseData,
+    "twoLetterCountryCode"
+  ).map(({ twoLetterCountryCode, data }) => ({
+    twoLetterCountryCode,
+    data: groupByArray(data, "year").map(({ year, data }) => ({
+      year,
+      data: groupByArray(data, "month"),
+    })),
+  }));
```

Comment:
I'm structuring it like this because it will make the lookups way more performant when it's time to associate.